lib/services/role.go

```diff
@@ -948,6 +948,42 @@ func (set RoleSet) EnumerateDatabaseUsers(database types.Database, extraUsers ..
 	return result
 }
 
+// EnumerateKubernetesUsersAndGroups works on a given role set to return a minimal list of
+// configured kubernetes_users and kubernetes_groups for a given kubernetes cluster.
+// The wildcard selector is ignored, since it is not allowed for kubernetes principals.
+func (set RoleSet) EnumerateKubernetesUsersAndGroups(kube types.KubeCluster) (EnumerationResult, EnumerationResult) {
```

Comment:
This seems very similar to what `CheckKubeGroupsAndUsers` does, which is what forwarder uses to determine which users/groups to pass. I understand the difference is that this operates on a particular cluster, but can't `CheckKubeGroupsAndUsers` be reused somehow? E.g. can't we pass it a matcher that would only match a particular cluster?

Reply:
You are correct. I didn't realize the presence of `CheckKubeGroupsAndUsers`. I changed the function and it now uses `CheckKubeGroupsAndUsers` internally with a label matcher.